plan/logical_plan_builder.go

```
@@ -231,6 +233,7 @@ func (b *planBuilder) buildJoin(join *ast.Join) LogicalPlan {
 
 	if b.TableHints() != nil {
 		joinPlan.preferMergeJoin = b.TableHints().ifPreferMergeJoin(leftAlias, rightAlias)
+		joinPlan.preferINLJ = b.TableHints().ifPreferINLJ(leftAlias, rightAlias)
```

If both preferMergeJoin and perferINLJ are true, we should report an error.